### DISCUSSION TITLE
Huggingface Download URL does not work

### DISCUSSION THREAD

**Discussion by moficodes** [mcmonkeyprojects/SwarmUI#184] [Q&A]
When I try to download a model safetensor from HuggingFace I get

```
Error: Failed to download https://huggingface.co/black-forest-labs/FLUX.1-dev/resolve/main/flux1-dev.safetensors: got response code 401 Unauthorized
Are you sure the URL is correct? Note some models may require you to authenticate using an [API Key](http://localhost:7801/Text2Image#).
```

I am not sure where to set the huggingface api key.

**Comment by moficodes** [comment]
This is the url I tried.

https://huggingface.co/black-forest-labs/FLUX.1-dev/blob/main/flux1-dev.safetensors

**Comment by mcmonkey4eva** [comment]
BFL Flux Dev has a gate on HF so you can't download it directly from the interface. It's also in "unet" format so it'll break anyway. Use the FP8 or NF4 model variants instead https://github.com/mcmonkeyprojects/SwarmUI/blob/master/docs/Model%20Support.md#black-forest-labs-flux1-models

**Reply by moficodes** [reply]
I should clarify, I did not manage to download any model from HF. 

Does SwarmUI have any way to take HF token as an env var? or set api token like stability or civit.

**Reply by mcmonkey4eva** [reply]
currently no, but, again: just download the the FP8 or NF4 versions of Flux, they will work fine without the need for any token. It's just very specifically the bfl flux dev model that needs a token, which also isn't going to work in the downloader util anyway because of the file format
